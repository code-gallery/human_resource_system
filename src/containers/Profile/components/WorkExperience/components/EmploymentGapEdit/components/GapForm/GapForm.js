import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'
import _isNil from 'lodash/isNil'
import { renderValidation } from 'containers/Profile/utils/validation'
import CustomDatePicker from 'components/FormElements/DatePicker'
import BaseForm from 'containers/Profile/components/AwardList/components/AwardItemEdit/components/Forms/BaseForm'
import { Document, Page, pdfjs } from 'react-pdf';
import "../style.css"
// pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

export default class GapForm extends BaseForm {
  constructor(props) {
    super(props);
    this.state = {type: "", 
    numPages: 0,
    pageNumber: 1
  }
  }

  
  onDocumentLoadSuccess = ({ numPages }) => {
    this.setState({ numPages });
  }
  componentWillMount() {
    this.reader = new FileReader()
    this.reader.addEventListener('load', this.onFileLoad)
  }

  componentWillUnmount() {
    this.reader.removeEventListener('load', this.onFileLoad)
  }


  onFileLoad = () => {
    this.props.onFieldChange('description', this.reader.result)
  
  }
  previousPage = () => this.changePage(-1);

  nextPage = () => this.changePage(1);

  changePage = offset => this.setState(prevState => ({
    pageNumber: (prevState.pageNumber || 1) + offset,
  }));

  handleFileChange = (event) => {
    const file = event.target.files[0]
    this.setState({type: file.type })
    if (file) {
      this.reader.readAsDataURL(file)
    }
  }

  render() {
    const {
      position,
      start_date,
      end_date,
      industry,
      reason,
      gapsReasons,
      invalidFields,
      invalidDates,
      invalidMessage,
      description,
      employment_type
    } = this.props
   
   const checkExt = (description && description.indexOf('https://') !== -1) ?
   description.split('.').pop() : null
   const { pageNumber, numPages, type } = this.state;
    return (
      <div>
         <div className="row">
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label className={renderValidation(invalidFields, 'employment_type')}>Employment Type</label>
              
              <Select
                name="select-content"
                options={[
                  {
                    value: 'employment_gap',
                    label: 'Employment Gap'
                  }
                ]}
                value={employment_type}
                onChange={v => this.handleChange(v ? v.value : '', 'employment_type')}
              />
            </div>
          </div>
        </div>
        <div className="row">
          
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label className={renderValidation(invalidFields, 'industry')}>Reason</label>
              <Select
                name="select-content"
                options={gapsReasons}
                labelKey="text"
                valueKey="text"
                value={{ text: industry || '' }}
                onChange={v => this.handleChange(v ? v.text : '', 'industry')}
              />
            </div>
          </div>
        </div>
        <div className="row">
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <CustomDatePicker
                handleDate={this.handleChange}
                invalid={invalidFields.indexOf('start_date') !== -1}
                date={start_date}
                title="start_date"
                label="Start Date"
              />
            </div>
          </div>
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <CustomDatePicker
                handleDate={this.handleChange}
                invalid={invalidFields.indexOf('end_date') !== -1}
                date={end_date}
                title="end_date"
                label="End Date"
                endDate
              />
            </div>
          </div>
          <div className="col-xs-12 col-md-6">
            <div className="form-group">
              <label>&nbsp;</label>
              <div className="btn-custom-file-upload">
                <label>
                  <input
                    type="file"
                    onChange={this.handleFileChange}
                    accept="application/pdf, image/jpeg"
                  />
                  <span>Upload supporting evidence (Optional)</span>
                </label>
              </div>
            </div>
          </div>
          </div>
        <div className="row">
          <div className="col-xs-12 col-md-12">
            
          {(description && type==='application/pdf' || checkExt==='pdf') ?(
            <div className='Test__container__content'>
              
              <Document file={description} 
              onLoadSuccess={this.onDocumentLoadSuccess} 
              className="custom-document-view"
              >
                <div className="Test__container__content__document"><Page pageNumber={pageNumber} /></div>
                <div className="Test__container__content__controls">
                  <button
                    disabled={pageNumber <= 1}
                    className={`prev-btn pull-left tooltip-position-top tooltip-movable tooltip-danger`}
                    onClick={this.previousPage}
                    data-tooltip="Previous Page"
                    type="button"
                  >
                     &nbsp;
                  </button>
                  <span>
                    {`Page ${pageNumber || (numPages ? 1 : '--')} of ${numPages || '--'}`}
                  </span>
                  <button
                    disabled={pageNumber >= numPages}
                    className={`next-btn pull-right tooltip-position-top tooltip-movable tooltip-danger`}
                    onClick={this.nextPage}
                    data-tooltip="Next Page"
                    type="button"
                  >
                      &nbsp;
                  </button>
                </div>
              </Document>
          
            
          </div>
          ):""}
           {(description && type==='image/jpeg' || checkExt==='jpg') ? (
              
            <div className="Test__container__content"><img src={description} alt="Evidence" /></div>
          ): ""}
        </div>
        </div>
        <div className="form-group">
          <label className={renderValidation(invalidFields, 'position')}>Details</label>
          <textarea rows="4" cols="50" name="position" value={position || ''} onChange={v => this.onTextChange('position', v)} />
        </div>
        {invalidDates && (<div className="required-error">{invalidMessage}</div>)}
      </div>
    )
  }
}

GapForm.propTypes = {
  
  position: PropTypes.string,
  start_date: PropTypes.string,
  end_date: PropTypes.string,
  industry: PropTypes.string,
  gapsReasons: PropTypes.array,
  invalidFields: PropTypes.array,
  invalidDates: PropTypes.bool,
  invalidMessage: PropTypes.string,
  employment_type: PropTypes.string
}
