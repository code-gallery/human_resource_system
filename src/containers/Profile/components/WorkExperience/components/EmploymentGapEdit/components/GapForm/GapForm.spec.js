import React from 'react'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme'
import GapForm from './index'

configure({adapter: new Adapter()});
describe('containers/Profile/components/.../GapForm', () => {
  const commonProps = {
    deleteWork: jest.fn(),
    orgRegistered: jest.fn(),
    onFieldChange: jest.fn(),
    saveEntity:jest.fn(),
    toggleEdit:jest.fn(),
    toggleBlockEditing:jest.fn(),
    invalidFields: [],
    invalidMessage: 'Error occured',
    invalidDates: false,
    industries: [
      { text: 'Accounting' },
      { text: 'Animation' },
      { text: 'IT' }
    ]
  }

  describe('when all inputs have no values', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps
      }
      const tree = renderer.create(
        <GapForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when invalidDates is true', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        invalidDates: true
      }
      const tree = renderer.create(
        <GapForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when "position" and "start_date" are invalid', () => {
    const props = {
      ...commonProps,
      industry: '',
      company:'Employment Gap',
      description: 'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
      end_date: '2016-03-01',
      start_date: '',
      position: 'Education Gap for an year ',
      invalidFields: [ 'industry', 'start_date' ]
    }

    it('renders correctly', () => {
      const tree = renderer.create(
        <GapForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when all inputs have values', () => {
    const props = {
      ...commonProps,
      employment_type: 'employment_gap',
      company:'Employment Gap',
      description: 'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
      industry: 'Compassionate Leave',
      end_date: '2016-06-01',
      start_date: '2016-03-01',
      position: 'Education Gap for an year'
    }

    it('renders correctly', () => {
      const tree = renderer.create(
        <GapForm {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    describe('handleChange', () => {
      it('calls props.onFieldChange correctly', () => {
        const wrapper = shallow(
          <GapForm {...props} />
        )
        props.onFieldChange.mockClear()
        wrapper.instance().handleChange('value', 'field')
        expect(props.onFieldChange)
          .toHaveBeenCalledWith('field', 'value')
      })
    })

    describe('onTextChange', () => {
      describe('when "industry" changes', () => {
        it('calls props.onFieldChange correctly', () => {
          const wrapper = shallow(
            <GapForm {...props} />
          )
          const event = { target: { name: 'industry', value: 'Compassionate Leave' } }
          props.onFieldChange.mockClear()
          wrapper.find('input[name="industry"]').simulate('change', event)
          expect(props.onFieldChange)
            .toHaveBeenCalledWith('industry', 'Compassionate Leave')
        })
      })
    })
  })
})
