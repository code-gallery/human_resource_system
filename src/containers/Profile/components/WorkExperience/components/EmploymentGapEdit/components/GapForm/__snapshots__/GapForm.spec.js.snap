// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/.../GapForm when "position" and "start_date" are invalid renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Employment Type
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-6--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-6--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className="required-error"
        >
          Reason
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value=""
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-7--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-7--value-item"
                  role="option"
                >
                  
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-7--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className="required-error"
          >
            Start Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              April 2018
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            End Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              March 2016
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label>
          Â 
        </label>
        <div
          className="btn-custom-file-upload"
        >
          <label>
            <input
              accept="application/pdf, image/jpeg"
              onChange={[Function]}
              type="file"
            />
            <span>
              Upload supporting evidence (Optional)
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-12"
    >
      <div
        className="Test__container__content"
      >
        Document
      </div>
      
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Details
    </label>
    <textarea
      cols="50"
      name="position"
      onChange={[Function]}
      rows="4"
      value="Education Gap for an year"
    />
  </div>
</div>
`;

exports[`containers/Profile/components/.../GapForm when all inputs have no values renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Employment Type
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-2--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-2--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Reason
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value=""
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-3--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-3--value-item"
                  role="option"
                >
                  
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-3--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            Start Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              April 2018
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            End Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              Present
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label>
          Â 
        </label>
        <div
          className="btn-custom-file-upload"
        >
          <label>
            <input
              accept="application/pdf, image/jpeg"
              onChange={[Function]}
              type="file"
            />
            <span>
              Upload supporting evidence (Optional)
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-12"
    >
      
      
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Details
    </label>
    <textarea
      cols="50"
      name="position"
      onChange={[Function]}
      rows="4"
      value=""
    />
  </div>
</div>
`;

exports[`containers/Profile/components/.../GapForm when all inputs have values renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Employment Type
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value="employment_gap"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-8--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-8--value-item"
                  role="option"
                >
                  Employment Gap
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-8--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Reason
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value="Compassionate Leave"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-9--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-9--value-item"
                  role="option"
                >
                  Compassionate Leave
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-9--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            Start Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              March 2016
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            End Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              June 2016
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label>
          Â 
        </label>
        <div
          className="btn-custom-file-upload"
        >
          <label>
            <input
              accept="application/pdf, image/jpeg"
              onChange={[Function]}
              type="file"
            />
            <span>
              Upload supporting evidence (Optional)
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-12"
    >
      <div
        className="Test__container__content"
      >
        Document
      </div>
      
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Details
    </label>
    <textarea
      cols="50"
      name="position"
      onChange={[Function]}
      rows="4"
      value="Education Gap for an year"
    />
  </div>
</div>
`;

exports[`containers/Profile/components/.../GapForm when invalidDates is true renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Employment Type
        </label>
        <div
          className="Select is-clearable is-searchable Select--single"
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-4--value"
            >
              <div
                className="Select-placeholder"
              >
                Select...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-4--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label
          className=""
        >
          Reason
        </label>
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="select-content"
            type="hidden"
            value=""
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="Select-multi-value-wrapper"
              id="react-select-5--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-5--value-item"
                  role="option"
                >
                  
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-5--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span
                className="Select-clear"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&times;",
                  }
                }
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            Start Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              April 2018
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <div
          className="form-group"
        >
          <label
            className=""
          >
            End Date
          </label>
          <div
            className="select double-select active"
            onClick={[Function]}
          >
            <span>
              Present
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-6"
    >
      <div
        className="form-group"
      >
        <label>
          Â 
        </label>
        <div
          className="btn-custom-file-upload"
        >
          <label>
            <input
              accept="application/pdf, image/jpeg"
              onChange={[Function]}
              type="file"
            />
            <span>
              Upload supporting evidence (Optional)
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-md-12"
    >
      
      
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Details
    </label>
    <textarea
      cols="50"
      name="position"
      onChange={[Function]}
      rows="4"
      value=""
    />
  </div>
  <div
    className="required-error"
  >
    Error occured
  </div>
</div>
`;
